domain/src/main/kotlin/no/nav/su/se/bakover/domain/behandling/Behandlingsinformasjon.kt

```diff
@@ -45,28 +45,27 @@ data class Behandlingsinformasjon(
         ektefelle = b.ektefelle ?: this.ektefelle,
     )
 
-    fun erInnvilget() = alleVilkår.all { it?.erVilkårOppfylt() ?: false }
+    fun erInnvilget() = alleVilkår.all { it !== null && it.erVilkårOppfylt() }
     fun getAvslagsgrunn() = alleVilkår.mapNotNull { it?.avslagsgrunn() }.firstOrNull()
     fun erAvslag(): Boolean {
-        val erFerdigbehandlet = alleVilkår.all { it != null && it.erGyldig() && it.erFerdigbehandlet() }
-
-        return uførhetOgFlyktningsstatusErVurdertOgMinstEnAvDeErIkkeOppfylt() || (erFerdigbehandlet && !erInnvilget())
+        return uførhetOgFlyktningsstatusErVurdertOgMinstEnAvDeErIkkeOppfylt() ||
+            (alleVilkår.all { it !== null } && alleVilkår.any { it!!.erVilkårIkkeOppfylt() })
     }
 
     private fun uførhetOgFlyktningsstatusErVurdertOgMinstEnAvDeErIkkeOppfylt(): Boolean {
         if (uførhet != null && flyktning != null) {
-            if (uførhet.status == Uførhet.Status.VilkårIkkeOppfylt || flyktning.status == Flyktning.Status.VilkårIkkeOppfylt) {
+            if (uførhet.erVilkårIkkeOppfylt() || flyktning.erVilkårIkkeOppfylt()) {
                 return true
             }
         }
         return false
     }
 
     abstract class Base {
-        open fun erGyldig(): Boolean = true
-        open fun erFerdigbehandlet(): Boolean = true
-        open fun erVilkårOppfylt(): Boolean = true
-        open fun avslagsgrunn(): Avslagsgrunn? = null
+        abstract fun erGyldig(): Boolean
```

Comment:
Jeg har ikke sjekka hele fila, men denne er true for alle tilfellene her.

Reply:
Ikke for uførhet, formue eller bosituasjon.